### DISCUSSION TITLE
Llama and Falcon in JAX

### DISCUSSION THREAD

**Discussion by rcrowe-google** [jax-ml/jax#17667] [General]
So far I've found at least 4 ports of Llama to JAX:

- Orig Llama: https://huggingface.co/heegyu/llama-7b-easylm
- Orig Llama: https://github.com/Sea-Snell/JAX_llama
- Llama2: https://github.com/ayaka14732/llama-2-jax
- Llama2: https://huggingface.co/erfanzar/Llama-2-jax

I'm wondering, are there are others that folks have done?

Also, I've looked for ports of Falcon to JAX and so far I haven't found any.  Do you know of any?

**Comment by davisyoshida** [comment]
Here's my haiku port of Llama: https://github.com/davisyoshida/llama-haiku

Haiku has perished though :broken_heart:

**Reply by rcrowe-google** [reply]
> Haiku has perished though 💔

Thanks for the link Davis, and sorry about Haiku.

**Comment by patrick-kidger** [comment]
Looks like you can find this in Levanter/Equinox: https://github.com/stanford-crfm/levanter/pull/270

**Reply by rcrowe-google** [reply]
Thanks Patrick!  Please let me know if you find a port of Falcon, I still haven't found one.

**Comment by ayaka14732** [comment]
Greetings from the author of llama-2-jax 👋🏻

**Comment by rcrowe-google** [comment]
Sorry, I forgot to include this one - https://github.com/openlm-research/open_llama

**Comment by defdet** [comment]
A bit late to the party, but there is falcon in jax now https://github.com/defdet/falcon-jax
